drivers/i2c/i2c_ite_enhance.c

```diff
@@ -256,7 +263,7 @@ static void i2c_enhanced_port_set_frequency(const struct device *dev,
 	 */
 	if (freq_hz) {
 		/* Get SMBus clock divide value */
-		clk_div = (IT8XXX2_ECPM_SCDCR2 & 0x0F) + 1U;
+		clk_div = (ecpm_base->SCDCR2 & 0x0F) + 1U;
```

Comment:
Move this logic into the clock control driver. `clock_control_get_rate(config->clk_dev, IT8XXX2_SMB_CLOCK, &smb_clock`).

Reply:
Thank you, sir. I'll try my best to move this logic into clock_control_it8xxx2 module.

Reply:
@keith-zephyr 

This part has been updated to get the current I2C source clock settings from the API clock_control_get_rate() as follows:

```
	struct it8xxx2_clkctrl_subsys_t subsys = {
		.clk_opt = IT8XXX2_SMB,
	};
		/* Get SMBus source clock value */
	clock_control_get_rate(config->clk_dev,
		(clock_control_subsys_t *)&subsys, &i2c_src_clock);
```
After that, the further calculation of the psr is as follows:

```
	if (freq_hz) {
		/* Calculate PSR value */
		psr = (i2c_src_clock / (2U * freq_hz)) - 2U;
```